### PR TITLE
Support inline comments

### PR DESCRIPTION
**Pull request by Lonami**
Closes https://github.com/Anuken/Mindustry-Suggestions/issues/1698.

### CODE REVIEW

File: core/src/mindustry/logic/LAssembler.java [hunk 1]
```diff
@@ -61,8 +61,14 @@ public static String write(Seq<LStatement> statements){
         String[] lines = data.split("\n");
         int index = 0;
         for(String line : lines){
-            //comments
-            if(line.startsWith("#") || line.isEmpty()) continue;
+            // comments
+            int commentIdx = line.indexOf('#');
+            if (commentIdx != -1) {
```
Comment (Quezler) [inline core/src/mindustry/logic/LAssembler.java:66]:
```suggestion
            if (commentIdx != -1){
```

File: core/src/mindustry/logic/LAssembler.java [hunk 2]
```diff
@@ -61,8 +61,14 @@ public static String write(Seq<LStatement> statements){
...
+                line = line.substring(0, commentIdx).trim();
+            }
+            if (line.isEmpty()) {
```
Comment (Quezler) [inline core/src/mindustry/logic/LAssembler.java:69]:
```suggestion
            if (line.isEmpty()){
```

**Review by Quezler — CHANGES_REQUESTED** [review]

File: core/src/mindustry/logic/LAssembler.java (hunk 1 above)
Comment (Lonami) [reply core/src/mindustry/logic/LAssembler.java:66]: Yeah, that's the "formatting aside" remark on the linked issue.

File: core/src/mindustry/logic/LAssembler.java [hunk 3]
```diff
@@ -61,8 +61,14 @@ public static String write(Seq<LStatement> statements){
...
-            //comments
-            if(line.startsWith("#") || line.isEmpty()) continue;
+            // comments
+            int commentIdx = line.indexOf('#');
+            if (commentIdx != -1){
+                line = line.substring(0, commentIdx).trim();
+            }
+            if (line.isEmpty()){
+                continue;
+            }
```
Comment (joshuaptfan) [inline core/src/mindustry/logic/LAssembler.java:71]: Make this a one-liner like in the code you replaced.
```suggestion
            if(line.isEmpty()) continue;
```
Also, in line 66, remove the space in `if (` to match Mindustry code style.

**Review by joshuaptfan — CHANGES_REQUESTED** [review]

**Pull request merged**